src/mesh/NodeDB.cpp

```diff
@@ -739,7 +739,18 @@ void NodeDB::updateFrom(const meshtastic_MeshPacket &mp)
 
         if (mp.rx_snr)
             info->snr = mp.rx_snr; // keep the most recent SNR we received for this node.
+
+        info->channel = mp.channel;
```

Comment:
consider saving NodeInfo.channel for NodeInfo(?) packets only, so that we only tag the remote nodes primary channel. 

otherwise it will save different channel values every time packets are received in any shared secondary channels.

Reply:
This is an improvement of cause, but it does not resolve channel-hopping entirely. what if 2 nodes are on 2 shared channels and a 3. node requests nodeinfo on a secondary. that will change the channel too since it hears a nodeinfo on a secondary channel. i was thinking about a channel change if a node is heard on a primary channel, or channel-index is higher in NodeDb than new packet. But there is my main problem again, how to differ between, heard on primary and unset.

Reply:
every node has only one primary channel. So if you trigger the update only on received nodeinfo packets you will always have that node's primary channel from your perspective. If this value is zero, you share a common primary channel. If the local value here is already non-zero, don't update it, cause you obviously hear the node across several paths. This assumes a fairly static setup.

Reply:
sorry, but i think that might not work if we send our nodeInfo on a secondary channel to reach out new unknown nodes while a third node is listening on that channel as well. I added that here: https://github.com/meshtastic/firmware/pull/2384/commits/85b18a4387383ad4e62f9e9a10393371872c53bd

Reply:
one option when receiving NodeInfo on a secondary channel --> `sendOurNodeInfo` with channel = our secondary channel (or any value for the matter, non-0 to indicate this is not our primary channel).

- nodes receiving a NodeInfo packet with a tagged NodeInfo.channel should consider lower priority 
- if we already seen that node on our primary don't replace (unless 2-3x `node_info_broadcast_secs` ago?)

thoughts?

Reply:
That's a good point - sholdn't we consider nodeinfo / nodedb stale at some point and clean it up? That's another issue/PR but it would take care of this situation. My suggestion to prioritize reception on a secondary channel over the primary was that it is easier to do in code. Here again 0 can mean 2 things...